src/lib/bootloader/autoyast_converter.rb

```diff
@@ -0,0 +1,90 @@
+require "yast"
+
+module Bootloader
+  # Converter between internal configuration model and autoyast serialization of configuration.
+  class AutoyastConverter
+    class << self
+      include Yast::Logger
+
+      def import(_data)
+      end
+
+      def export(config)
+        log.info "exporting config #{config.inspect}"
+
+        bootloader_type = config.name
+        res = { "loader_type" => bootloader_type }
+
+        return res if bootloader_type == "none"
+
+        export_stage1(res, config.stage1) if config.respond_to?(:stage1)
+        export_default(res, config.grub_default)
+
+        res
+      end
+
+    private
+
+      STAGE1_MAPPING = {
+        "activate"      => :activate?,
+        "generic_mbr"   => :generic_mbr?,
+        "boot_root"     => :root_partition?,
+        "boot_boot"     => :boot_partition?,
+        "boot_mbr"      => :mbr?,
+        "boot_extended" => :extended_partition?
+      }
+      def export_stage1(res, stage1)
+        res["global"] ||= {}
+        STAGE1_MAPPING.each do |key, method|
+          res["global"][key] = stage1.public_send(method) ? "true" : "false"
+        end
+
+        res["global"]["boot_custom"] = stage1.custom_devices.join(",") unless stage1.custom_devices.empty?
+      end
+
+      DEFAULT_BOOLEAN_MAPPING = {
+        "os_prober" => :os_prober
+      }
+
+      DEFAULT_STRING_MAPPING = {
+        "gfxmode"  => :gfxmode,
+        "serial"   => :serial_console,
+        "terminal" => :terminal
+      }
+
+      DEFAULT_KERNEL_PARAMS_MAPPING = {
+        "append"            => :kernel_params,
+        "xen_append"        => :xen_kernel_params,
+        "xen_kernel_append" => :xen_hypervisor_params
+      }
+      def export_default(res, default)
+        DEFAULT_BOOLEAN_MAPPING.each do |key, method|
+          val = default.public_send(method)
+          res["global"][key] = val.enabled? ? "true" : "false" if val.defined?
```

Comment:
yes, used only here and to be more funny it is string and not boolean xml attribute due to old p-bl map which use strings.